app/controllers/discourse_ai/ai_helper/assistant_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module DiscourseAi
+  module AiHelper
+    class AssistantController < ::ApplicationController
+      requires_plugin ::DiscourseAi::PLUGIN_NAME
+      requires_login
+      before_action :ensure_can_request_suggestions
+
+      def suggest
+        raise Discourse::InvalidParameters.new(:text) if params[:text].blank?
+
+        if !DiscourseAi::AiHelper::OpenAiPrompt::VALID_TYPES.include?(params[:mode])
+          raise Discourse::InvalidParameters.new(:mode)
+        end
+
+        RateLimiter.new(current_user, "ai_assistant", 6, 3.minutes).performed!
```

Comment:
Maybe bypass for staff? Specially now that everyone will be playing with it

Reply:
Staff is bypassed by default unless specified otherwise